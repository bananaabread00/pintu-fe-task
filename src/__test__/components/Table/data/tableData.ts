import { HeadCell } from '../../../../interfaces/table';

interface DessertList {
  name: string;
  calories: number;
  fat: number;
}

export const headCells: HeadCell<DessertList>[] = [
  {
    id: 'name',
    label: 'Dessert'
  },
  {
    id: 'calories',
    label: 'Calories',
    sortable: true
  },
  {
    id: 'fat',
    label: 'Fat (g)',
    sortable: true,
    width: 250
  }
];

export const tableData = [
  {
    name: 'Frozen yoghurt',
    calories: 100,
    fat: 80
  },
  {
    name: 'Ice cream',
    calories: 180,
    fat: 97
  },
  {
    name: 'Eclair',
    calories: 235,
    fat: 98
  }
];
