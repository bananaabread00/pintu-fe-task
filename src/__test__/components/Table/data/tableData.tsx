import React from 'react';
import { HeadCell } from '../../../../interfaces/table';

interface DessertList {
  name: string;
  calories: number;
  fat: number;
}

export const sortableHeadCells: HeadCell<DessertList>[] = [
  {
    id: 'name',
    label: 'Dessert',
    render: (value) => <strong>{value}</strong>
  },
  {
    id: 'calories',
    label: 'Calories',
    sortable: true
  },
  {
    id: 'fat',
    label: 'Fat (g)',
    sortable: true,
    width: 250
  }
];

export const dropdownHeadCells: HeadCell<DessertList>[] = [
  {
    id: 'name',
    label: 'Dessert',
    main: true
  },
  {
    id: 'calories',
    label: 'Calories'
  }
];

export const tableData = [
  {
    name: 'Frozen yoghurt',
    calories: 100,
    fat: 80
  },
  {
    name: 'Ice cream',
    calories: 180,
    fat: 97
  },
  {
    name: 'Eclair',
    calories: 235,
    fat: 98
  }
];

export const dropdownList = [
  { key: 'calories', label: 'Calories' },
  { key: 'fat', label: 'Fat (g)' }
];
