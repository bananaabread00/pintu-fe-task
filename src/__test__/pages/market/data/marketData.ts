export const getSupportedCurrenciesData = {
  code: 'success',
  payload: [
    {
      currencyGroup: 'BTC',
      color: '#F78B1A',
      currencySymbol: 'BTC',
      name: 'Bitcoin',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_BTC.svg',
      decimal_point: 8,
      listingDate: '2020-09-15T09:43:45Z',
      wallets: [
        {
          currencyGroup: 'BTC',
          tokenSymbol: 'BTC',
          decimal_point: 8,
          tokenType: 'Bitcoin',
          blockchain: 'Bitcoin',
          explorer: 'https://explorer.bitcoin.com/btc/tx/',
          listingDate: '2020-09-15T09:43:45Z',
          blockchainName: 'Bitcoin',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/Bitcoin.svg'
        }
      ]
    },
    {
      currencyGroup: 'ETH',
      color: '#9011FE',
      currencySymbol: 'ETH',
      name: 'Ethereum',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_ETH.svg',
      decimal_point: 18,
      listingDate: '2020-09-15T09:43:46Z',
      wallets: [
        {
          currencyGroup: 'ETH',
          tokenSymbol: 'ETH',
          decimal_point: 18,
          tokenType: 'ERC-20',
          blockchain: 'Ethereum',
          explorer: 'https://etherscan.io/tx/',
          listingDate: '2020-09-15T09:43:46Z',
          blockchainName: 'Ethereum',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/ERC-20.svg'
        }
      ]
    },
    {
      currencyGroup: 'USDT',
      color: '#15A57C',
      currencySymbol: 'USDT',
      name: 'Tether',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_USDT.svg',
      decimal_point: 6,
      listingDate: '2020-09-15T09:43:47Z',
      wallets: [
        {
          currencyGroup: 'USDT',
          tokenSymbol: 'USDT',
          decimal_point: 6,
          tokenType: 'ERC-20',
          blockchain: 'Ethereum',
          explorer: 'https://etherscan.io/tx/',
          listingDate: '2020-09-15T09:43:47Z',
          blockchainName: 'Ethereum',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/ERC-20.svg'
        }
      ]
    },
    {
      currencyGroup: 'BNB',
      color: '#FEBF11',
      currencySymbol: 'BNB',
      name: 'Binance Coin',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_BNB.svg',
      decimal_point: 8,
      listingDate: '2020-09-15T09:43:48Z',
      wallets: [
        {
          currencyGroup: 'BNB',
          tokenSymbol: 'BNB',
          decimal_point: 8,
          tokenType: 'BEP-2',
          blockchain: 'Binance',
          explorer: 'https://explorer.binance.org/tx/',
          listingDate: '2020-09-15T09:43:48Z',
          blockchainName: 'BNB Beacon Chain',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/BEP-2.svg'
        }
      ]
    }
  ],
  message: ''
};

export const getPriceChangesData = {
  code: 'success',
  payload: [
    {
      pair: 'btc/idr',
      latestPrice: '10268',
      day: '-4.84',
      week: '-7.39',
      month: '-7.64',
      year: '-24.52'
    },
    {
      pair: 'eth/idr',
      latestPrice: '21904',
      day: '-0.89',
      week: '-8.15',
      month: '-5.24',
      year: '-54.20'
    },
    {
      pair: 'usdt/idr',
      latestPrice: '28825405',
      day: '0.09',
      week: '1.32',
      month: '0.95',
      year: '11.00'
    },
    {
      pair: 'bnb/idr',
      latestPrice: '104711228',
      day: '-3.15',
      week: '-4.57',
      month: '18.85',
      year: '7.83'
    }
  ],
  message: ''
};
