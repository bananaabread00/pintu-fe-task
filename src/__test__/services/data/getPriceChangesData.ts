export const getPriceChangesData = {
  code: 'success',
  payload: [
    {
      pair: 'btc/idr',
      latestPrice: '10268',
      day: '-4.84',
      week: '-7.39',
      month: '-7.64',
      year: '-24.52'
    },
    {
      pair: 'eth/idr',
      latestPrice: '21904',
      day: '-0.89',
      week: '-8.15',
      month: '-5.24',
      year: '-54.20'
    },
    {
      pair: 'usdt/idr',
      latestPrice: '28825405',
      day: '0.09',
      week: '1.32',
      month: '0.95',
      year: '11.00'
    },
    {
      pair: 'bnb/idr',
      latestPrice: '104711228',
      day: '-3.15',
      week: '-4.57',
      month: '18.85',
      year: '7.83'
    }
  ],
  message: ''
};
