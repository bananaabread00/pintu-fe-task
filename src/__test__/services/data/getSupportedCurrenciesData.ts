export const getSupportedCurrenciesData = {
  code: 'success',
  payload: [
    {
      currencyGroup: 'BTC',
      color: '#F78B1A',
      currencySymbol: 'BTC',
      name: 'Bitcoin',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_BTC.svg',
      decimal_point: 8,
      listingDate: '2020-09-15T09:43:45Z',
      wallets: [
        {
          currencyGroup: 'BTC',
          tokenSymbol: 'BTC',
          decimal_point: 8,
          tokenType: 'Bitcoin',
          blockchain: 'Bitcoin',
          explorer: 'https://explorer.bitcoin.com/btc/tx/',
          listingDate: '2020-09-15T09:43:45Z',
          blockchainName: 'Bitcoin',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/Bitcoin.svg'
        }
      ]
    },
    {
      currencyGroup: 'ETH',
      color: '#9011FE',
      currencySymbol: 'ETH',
      name: 'Ethereum',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_ETH.svg',
      decimal_point: 18,
      listingDate: '2020-09-15T09:43:46Z',
      wallets: [
        {
          currencyGroup: 'ETH',
          tokenSymbol: 'ETH',
          decimal_point: 18,
          tokenType: 'ERC-20',
          blockchain: 'Ethereum',
          explorer: 'https://etherscan.io/tx/',
          listingDate: '2020-09-15T09:43:46Z',
          blockchainName: 'Ethereum',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/ERC-20.svg'
        }
      ]
    },
    {
      currencyGroup: 'USDT',
      color: '#15A57C',
      currencySymbol: 'USDT',
      name: 'Tether',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_USDT.svg',
      decimal_point: 6,
      listingDate: '2020-09-15T09:43:47Z',
      wallets: [
        {
          currencyGroup: 'USDT',
          tokenSymbol: 'USDT',
          decimal_point: 6,
          tokenType: 'ERC-20',
          blockchain: 'Ethereum',
          explorer: 'https://etherscan.io/tx/',
          listingDate: '2020-09-15T09:43:47Z',
          blockchainName: 'Ethereum',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/ERC-20.svg'
        }
      ]
    },
    {
      currencyGroup: 'BNB',
      color: '#FEBF11',
      currencySymbol: 'BNB',
      name: 'Binance Coin',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_BNB.svg',
      decimal_point: 8,
      listingDate: '2020-09-15T09:43:48Z',
      wallets: [
        {
          currencyGroup: 'BNB',
          tokenSymbol: 'BNB',
          decimal_point: 8,
          tokenType: 'BEP-2',
          blockchain: 'Binance',
          explorer: 'https://explorer.binance.org/tx/',
          listingDate: '2020-09-15T09:43:48Z',
          blockchainName: 'BNB Beacon Chain',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/BEP-2.svg'
        }
      ]
    }
  ],
  message: ''
};
