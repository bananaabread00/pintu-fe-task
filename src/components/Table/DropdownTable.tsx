import React from 'react';
import Box from '@mui/material/Box';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Paper from '@mui/material/Paper';
import { DropdownTableProps, EnhancedDropdownTableProps } from '../../interfaces/table';
import Select, { SelectChangeEvent } from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';
import FormControl from '@mui/material/FormControl';
import { useState } from 'react';

function EnhancedTableHead<T>(props: EnhancedDropdownTableProps<T>) {
  const { headCells, dropdownList, onSelectionChange } = props;
  const [selection, setSelection] = useState<string>(dropdownList[0].key);

  const handleChange = (event: SelectChangeEvent) => {
    setSelection(event.target.value);
    onSelectionChange(event.target.value);
  };

  return (
    <TableHead>
      <TableRow>
        {headCells.map((headCell) => (
          <TableCell
            align={headCell.align || 'left'}
            key={headCell.id as string}
            sx={{ py: 4, fontSize: '1rem' }}
          >
            {headCell.main ? (
              <span className="text-gray-600 text-center">{headCell.label}</span>
            ) : (
              <FormControl>
                <Select value={selection} displayEmpty size="small" onChange={handleChange}>
                  {dropdownList.map((item) => (
                    <MenuItem key={item.key} value={item.key}>
                      {item.label}
                    </MenuItem>
                  ))}
                </Select>
              </FormControl>
            )}
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
}

function DropdownTable<T>(props: DropdownTableProps<T>) {
  const { headCells, tableData, dropdownList, onSelectionChange } = props;

  const defaultRender = (value: any) => <span>{value}</span>;

  return (
    <Box sx={{ width: '100%' }}>
      <Paper sx={{ width: '100%', mb: 2 }}>
        <TableContainer sx={{ maxHeight: '85vh' }}>
          <Table stickyHeader aria-labelledby="tableTitle" size="medium">
            <EnhancedTableHead
              headCells={headCells}
              dropdownList={dropdownList}
              onSelectionChange={onSelectionChange}
            />
            <TableBody>
              {tableData.map((row, index) => (
                <TableRow key={index}>
                  {headCells.map((column) => (
                    <TableCell
                      key={column.id as string}
                      sx={{ py: 3, fontSize: '1rem', width: column.width }}
                      align={column.align || 'left'}
                    >
                      {column.render
                        ? column.render(row[column.id], row)
                        : defaultRender(row[column.id])}
                    </TableCell>
                  ))}
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>
      </Paper>
    </Box>
  );
}

export default DropdownTable;
