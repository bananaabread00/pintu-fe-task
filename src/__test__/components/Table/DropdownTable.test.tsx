import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import DropdownTable from '../../../components/Table/DropdownTable';
import { dropdownList, dropdownHeadCells, tableData } from './data/tableData';
import userEvent from '@testing-library/user-event';

describe('SortableTable component test', () => {
  it('should render the table', async () => {
    const { container } = render(
      <DropdownTable
        headCells={dropdownHeadCells}
        tableData={tableData}
        dropdownList={dropdownList}
        onSelectionChange={() => null}
      />
    );
    expect(await screen.findByText('Calories')).toBeInTheDocument();
    expect(container.getElementsByClassName('MuiTableRow-root').length).toBe(4);
    expect(container).toMatchSnapshot();
  });
  it('should sort the table data', async () => {
    const logSpy = jest.spyOn(console, 'log');
    const { container, getAllByRole } = render(
      <DropdownTable
        headCells={dropdownHeadCells}
        tableData={tableData}
        dropdownList={dropdownList}
        onSelectionChange={(value) => console.log('selected value:', value)}
      />
    );
    const [dropdown] = getAllByRole('button');
    await userEvent.click(dropdown);
    const [, option2] = getAllByRole('option');
    await userEvent.click(option2);
    expect(await screen.findByText('Fat (g)')).toBeInTheDocument();
    expect(logSpy).toHaveBeenCalledWith('selected value:', 'fat');

    expect(container).toMatchSnapshot();
  });
});
