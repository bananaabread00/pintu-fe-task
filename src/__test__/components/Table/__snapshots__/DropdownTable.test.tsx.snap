// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortableTable component test should render the table 1`] = `
<div>
  <div
    class="MuiBox-root css-8atqhb"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1mxz8qt-MuiPaper-root"
    >
      <div
        class="MuiTableContainer-root css-vq2emh-MuiTableContainer-root"
      >
        <table
          aria-labelledby="tableTitle"
          class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium css-10sdfzr-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="text-gray-600 text-center"
                >
                  Dessert
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium css-10sdfzr-MuiTableCell-root"
                scope="col"
              >
                <div
                  class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                      role="button"
                      tabindex="0"
                    >
                      Calories
                    </div>
                    <input
                      aria-hidden="true"
                      aria-invalid="false"
                      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                      tabindex="-1"
                      value="calories"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-ihdtdm"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  Frozen yoghurt
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  100
                </span>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  Ice cream
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  180
                </span>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  Eclair
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  235
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`SortableTable component test should sort the table data 1`] = `
<div>
  <div
    class="MuiBox-root css-8atqhb"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1mxz8qt-MuiPaper-root"
    >
      <div
        class="MuiTableContainer-root css-vq2emh-MuiTableContainer-root"
      >
        <table
          aria-labelledby="tableTitle"
          class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium css-10sdfzr-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="text-gray-600 text-center"
                >
                  Dessert
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium css-10sdfzr-MuiTableCell-root"
                scope="col"
              >
                <div
                  class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                      role="button"
                      tabindex="0"
                    >
                      Fat (g)
                    </div>
                    <input
                      aria-hidden="true"
                      aria-invalid="false"
                      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                      tabindex="-1"
                      value="fat"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-ihdtdm"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  Frozen yoghurt
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  100
                </span>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  Ice cream
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  180
                </span>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  Eclair
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  235
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
